### PR TITLE
Features/manual poke coin collection

### PR DESCRIPTION
**Pull request by Lord-Haji**

### CODE REVIEW

**Comment by Furtif** [comment]
fix this https://github.com/Necrobot-Private/NecroBot/pull/1767

**Comment by CDAGaming** [comment]
@Furtif Taking care of this already

**Comment by Furtif** [comment]
merge this https://github.com/Necrobot-Private/NecroBot/pull/1767

**Comment by CDAGaming** [comment]
Making an adjusted PR until Haji Updates his

**Comment by Furtif** [comment]
odd https://github.com/Necrobot-Private/NecroBot/compare/master...Furtif:features/ManualPokeCoinCollection

**Comment by CDAGaming** [comment]
I hope you've kept in mind #1768, as it included some layout tweaks to that button. If not, I can't reapply until tonight

**Comment by Furtif** [comment]
yes https://github.com/Necrobot-Private/NecroBot/pull/1776 ready

**Comment by Furtif** [comment]
merge #1779

**Comment by CDAGaming** [comment]
INSPECTION!

Going to be inspecting the changes and working status, and removing/simplifying unnecessary diffs 👍

File: PoGo.NecroBot.CLI/Config/Translations/translation.fr.json [hunk 1]
```diff
@@ -126,11 +126,11 @@
     },
     {
       "Key": "eventPokemonCaptureSuccess",
-      "Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | dist {8}m | avec une {9} ({10} restante(s)). | {11} EXP | {12} | lat: {13} long: {14} | Attaque1: {15} Attaque2: {16} | Rareté: {17} | Raison de capture: {18} | Shiny: {19} | Form: {20} | Costume: {21} |  Sex: {22}"
+      "Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | dist {8}m | avec une {9} ({10} restante(s)). | XP: {11} | SD: {12} | {13} | lat: {14} long: {15} | Attaque1: {16} Attaque2: {17} | Rareté: {18} | Raison de capture: {19} | Shiny: {20} | Form: {21} | Costume: {22} | Sex: {23}"
     },
     {
       "Key": "eventPokemonCaptureFailed",
-      "Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | {8}m dist | avec une {9} ({10} restante(s)) | lat: {11} long: {12} | Attaque1: {13} Attaque2: {14}  | Rareté :{15}"
+      "Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | {8}m dist | avec une {9} ({10}  restante(s)). | XP: {11} | lat: {12} long: {13} | Attaque1: {14} Attaque2: {15} | Rareté :{16}"
```
Comment (CDAGaming) [inline PoGo.NecroBot.CLI/Config/Translations/translation.fr.json]: Attacae 1 and 2 would need a '|' to split them

File: PoGo.NecroBot.CLI/Program.cs [hunk 2]
```diff
@@ -475,11 +475,8 @@ public static void RunBotWithParameters(Action<ISession, StatisticsAggregator> o
             {
             }
 
-            Logger.Write(
-               $"(Start-Up Stats) User: {bot.Username} | XP: {bot.CurrentXp} | SD: {bot.Stardust}",
-               LogLevel.Info, ConsoleColor
-               .Magenta
-               );
+            Logger.Write($"(Start-Up Stats) User: {bot.Username} | XP: {bot.CurrentXp} | SD: {bot.Stardust}",
```
Comment (CDAGaming) [inline PoGo.NecroBot.CLI/Program.cs]: Can be reduced to one line (Optional)

File: PoGo.NecroBot.Logic/Tasks/CatchNearbyPokemonsTask.cs [hunk 3]
```diff
@@ -101,8 +101,17 @@ public static class CatchNearbyPokemonsTask
...
             }
-            
-            if (0 < pokemons.Count)
+
+            var allitems = await session.Inventory.GetItems().ConfigureAwait(false);
+            var pokeBallsCount = allitems.FirstOrDefault(i => i.ItemId == ItemId.ItemPokeBall)?.Count;
+            var greatBallsCount = allitems.FirstOrDefault(i => i.ItemId == ItemId.ItemGreatBall)?.Count;
+            var ultraBallsCount = allitems.FirstOrDefault(i => i.ItemId == ItemId.ItemUltraBall)?.Count;
+            var masterBallsCount = allitems.FirstOrDefault(i => i.ItemId == ItemId.ItemMasterBall)?.Count;
+            masterBallsCount =
```
Comment (CDAGaming) [inline PoGo.NecroBot.Logic/Tasks/CatchNearbyPokemonsTask.cs]: If i can remember, i did a calling similar to this during SuperBuild, and it gave off errors in build.bat. If it runs successfully in build.bat, this comment can be ignored.

File: PoGo.NecroBot.Logic/Tasks/CatchNearbyPokemonsTask.cs [hunk 4]
```diff
@@ -125,15 +134,7 @@ public static class CatchNearbyPokemonsTask
...
-                var allitems = await session.Inventory.GetItems().ConfigureAwait(false);
-                var pokeBallsCount = allitems.FirstOrDefault(i => i.ItemId == ItemId.ItemPokeBall)?.Count;
-                var greatBallsCount = allitems.FirstOrDefault(i => i.ItemId == ItemId.ItemGreatBall)?.Count;
-                var ultraBallsCount = allitems.FirstOrDefault(i => i.ItemId == ItemId.ItemUltraBall)?.Count;
-                var masterBallsCount = allitems.FirstOrDefault(i => i.ItemId == ItemId.ItemMasterBall)?.Count;
-                masterBallsCount =
-                    masterBallsCount ?? 0; //return null ATM. need this code to logic check work
-
-                if (pokeBallsCount + greatBallsCount + ultraBallsCount + masterBallsCount <
+                if (PokeBalls <
```
Comment (CDAGaming) [inline PoGo.NecroBot.Logic/Tasks/CatchNearbyPokemonsTask.cs]: Can be simplified to one line, if removing extra indents

File: PoGo.NecroBot.Logic/Tasks/EvolvePokemonTask.cs [hunk 5]
```diff
@@ -123,7 +151,9 @@ public class EvolvePokemonTask
                     try
                     {
                         // no cancellationToken.ThrowIfCancellationRequested here, otherwise the lucky egg would be wasted.
-                        var evolveResponse = await session.Client.Inventory.EvolvePokemon(pokemon.Id ,filter== null? ItemId.ItemUnknown: await GetRequireEvolveItem(session ,pokemon.PokemonId, filter.EvolveToPokemonId)).ConfigureAwait(false);
```
Comment (CDAGaming) [inline PoGo.NecroBot.Logic/Tasks/EvolvePokemonTask.cs]: Unneccesary Change( Should be 'TextHere, MoreText' not 'TextHere ,MoreText' )

File: PoGo.Necrobot.Window/Controls/PlayerInfo.xaml [hunk 6]
```diff
@@ -110,9 +98,11 @@
                     </MultiBinding>
                 </TextBlock.Text>
             </TextBlock>
-
-
+            <TextBlock Grid.Column="2" Grid.Row="2" x:Name="tb_lukyegg_time" TextWrapping="Wrap" Text="{Binding Lucky_expires, FallbackValue=LuckyEggs: 00m 00s, StringFormat=LuckyEggs: \{0\}, UpdateSourceTrigger=PropertyChanged}" VerticalAlignment="Top" Width="158" Height="16"/>
+            <TextBlock Grid.Column="3" Grid.Row="2" x:Name="tb_incense_time" TextWrapping="Wrap" Text="{Binding Insence_expires, FallbackValue=Insence: 00m 00s, StringFormat=Insence: \{0\}, UpdateSourceTrigger=PropertyChanged}" VerticalAlignment="Top" Width="158" Height="16"/>
+            <Button x:Name="btnCollectPokeCoin" Click="GetPokeCoin_Click" Grid.Column="4" Grid.Row="1" HorizontalAlignment="Center" VerticalAlignment="Center" Margin="23.4,24.4,24,0.2" Grid.RowSpan="2" Width="111">
```
Comment (CDAGaming) [inline PoGo.Necrobot.Window/Controls/PlayerInfo.xaml]: For this Button, Margin wouldn't really be necessary...

Comment (CDAGaming) [inline PoGo.Necrobot.Window/Controls/PlayerInfo.xaml]: I also dont see why 2 Grid Rows would be needed, unless pokecoins go over 9 Digits

File: PoGo.Necrobot.Window/Model/PlayerInfoModel.cs [hunk 7]
```diff
@@ -1,20 +1,51 @@
...
+using PoGo.NecroBot.Logic.State;
+using POGOProtos.Inventory.Item;
+using System.Timers;
 
 namespace PoGo.Necrobot.Window.Model
 {
     public class PlayerInfoModel : ViewModelBase
     {
+        private DateTime expires = new DateTime(0);
+        private System.Timers.Timer tmr = new Timer()
```
Comment (CDAGaming) [inline PoGo.Necrobot.Window/Model/PlayerInfoModel.cs]: In Line 13 System.Timers is already defined, so i don't see it necessary to be referenced twice

File: RocketBot2/Config/Translations/translation.fr.json [hunk 8]
```diff
@@ -126,11 +126,11 @@
     },
     {
       "Key": "eventPokemonCaptureSuccess",
-      "Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | dist {8}m | avec une {9} ({10} restante(s)). | {11} EXP | {12} | lat: {13} long: {14} | Attaque1: {15} Attaque2: {16} | Rareté: {17} | Raison de capture: {18} | Shiny: {19} | Form: {20} | Costume: {21} |  Sex: {22}"
+      "Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | dist {8}m | avec une {9} ({10} restante(s)). | XP: {11} | SD: {12} | {13} | lat: {14} long: {15} | Attaque1: {16} Attaque2: {17} | Rareté: {18} | Raison de capture: {19} | Shiny: {20} | Form: {21} | Costume: {22} | Sex: {23}"
     },
     {
       "Key": "eventPokemonCaptureFailed",
-      "Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | {8}m dist | avec une {9} ({10} restante(s)) | lat: {11} long: {12} | Attaque1: {13} Attaque2: {14}  | Rareté :{15}"
+      "Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | {8}m dist | avec une {9} ({10}  restante(s)). | XP: {11} | lat: {12} long: {13} | Attaque1: {14} Attaque2: {15} | Rareté :{16}"
```
Comment (CDAGaming) [inline RocketBot2/Config/Translations/translation.fr.json]: Same Review Statements recommended as Necrobot Translation File for 'fr'

File: RocketBot2/Forms/MainForm.cs [hunk 9]
```diff
@@ -95,6 +95,9 @@ public MainForm(string[] _args)
 
         private void MainForm_Load(object sender, EventArgs e)
         {
+            this.splitContainer1.SplitterDistance = this.splitContainer1.Width / 100 * 45; // Splits left & right splitter panes @ 45%/55% of the window width
```
Comment (CDAGaming) [inline RocketBot2/Forms/MainForm.cs]: Extra Spaces on Width / 100 and Height / 100 are not neccesary

File: RocketBot2/Models/GMapMarkerPokestops.cs [hunk 10]
```diff
@@ -30,7 +30,7 @@ public GMapMarkerPokestops(PointLatLng p, Image image)
         {
             MarkerImage = image;
             Size = MarkerImage.Size;
-            Offset = new Point(-Size.Width/2, -Size.Height);
+            Offset = new Point(-Size.Width / 2, -Size.Height / 2);
```
Comment (CDAGaming) [inline RocketBot2/Models/GMapMarkerPokestops.cs]: Excess Space with Width / 2 and Height / 2

File: RocketBot2/Models/GMapMarkerTrainer.cs [hunk 11]
```diff
@@ -30,7 +30,7 @@ public GMapMarkerTrainer(PointLatLng p, Image image)
         {
             MarkerImage = image;
             Size = MarkerImage.Size;
-            Offset = new Point(-Size.Width/2, -Size.Height/2);
+            Offset = new Point(-Size.Width / 2, -Size.Height / 2);
```
Comment (CDAGaming) [inline RocketBot2/Models/GMapMarkerTrainer.cs]: Excess Space with Width / 2 and Height / 2

**Comment by CDAGaming** [comment]
INSPECTION AND REVIEW COMPLETE!

**Comment by TheWizard1328** [comment]
See my comments in red...

In PoGo.NecroBot.CLI/Config/Translations/translation.fr.json<https://github.com/Necrobot-Private/NecroBot/pull/1757#discussion_r119266581>:

>      },

     {

       "Key": "eventPokemonCaptureFailed",

-      "Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | {8}m dist | avec une {9} ({10} restante(s)) | lat: {11} long: {12} | Attaque1: {13} Attaque2: {14}  | Rareté :{15}"

+      "Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | {8}m dist | avec une {9} ({10}  restante(s)). | XP: {11} | lat: {12} long: {13} | Attaque1: {14} Attaque2: {15} | Rareté :{16}"

I dont think restante(s) needs an extra space or a period after.

I only edited the English translations. I see your logic here too.


In PoGo.NecroBot.CLI/Config/Translations/translation.fr.json<https://github.com/Necrobot-Private/NecroBot/pull/1757#discussion_r119266675>:

>      },

     {

       "Key": "eventPokemonCaptureFailed",

-      "Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | {8}m dist | avec une {9} ({10} restante(s)) | lat: {11} long: {12} | Attaque1: {13} Attaque2: {14}  | Rareté :{15}"

+      "Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | {8}m dist | avec une {9} ({10}  restante(s)). | XP: {11} | lat: {12} long: {13} | Attaque1: {14} Attaque2: {15} | Rareté :{16}"

Attacae 1 and 2 would need a '|' to split them

Same here as above.


In PoGo.NecroBot.CLI/Program.cs<https://github.com/Necrobot-Private/NecroBot/pull/1757#discussion_r119266757>:

> @@ -475,11 +475,8 @@ public static void RunBotWithParameters(Action<ISession, StatisticsAggregator> o

             {

             }



-            Logger.Write(

-               $"(Start-Up Stats) User: {bot.Username} | XP: {bot.CurrentXp} | SD: {bot.Stardust}",

-               LogLevel.Info, ConsoleColor

-               .Magenta

-               );

+            Logger.Write($"(Start-Up Stats) User: {bot.Username} | XP: {bot.CurrentXp} | SD: {bot.Stardust}",

Can be reduced to one line (Optional)

I am already working on this. Tested it and noticed there were duplicated logs. Update forth coming soon.


In PoGo.NecroBot.Logic/Tasks/CatchNearbyPokemonsTask.cs<https://github.com/Necrobot-Private/NecroBot/pull/1757#discussion_r119266860>:

> @@ -101,8 +101,17 @@ public static class CatchNearbyPokemonsTask

                     }

                 }

             }

-

-            if (0 < pokemons.Count)

+

+            var allitems = await session.Inventory.GetItems().ConfigureAwait(false);

+            var pokeBallsCount = allitems.FirstOrDefault(i => i.ItemId == ItemId.ItemPokeBall)?.Count;

+            var greatBallsCount = allitems.FirstOrDefault(i => i.ItemId == ItemId.ItemGreatBall)?.Count;

+            var ultraBallsCount = allitems.FirstOrDefault(i => i.ItemId == ItemId.ItemUltraBall)?.Count;

+            var masterBallsCount = allitems.FirstOrDefault(i => i.ItemId == ItemId.ItemMasterBall)?.Count;

+            masterBallsCount =

If i can remember, i did a calling similar to this during SuperBuild, and it gave off errors in build.bat. If it runs successfully in build.bat, this comment can be ignored.

This is not my code. But I did move it up a few lines to utilize it earlier/sooner.



In PoGo.NecroBot.Logic/Tasks/CatchNearbyPokemonsTask.cs<https://github.com/Necrobot-Private/NecroBot/pull/1757#discussion_r119266913>:

> @@ -125,15 +134,7 @@ public static class CatchNearbyPokemonsTask

                     continue; //this pokemon has been skipped because not meet with catch criteria before.

                 }



-                var allitems = await session.Inventory.GetItems().ConfigureAwait(false);

-                var pokeBallsCount = allitems.FirstOrDefault(i => i.ItemId == ItemId.ItemPokeBall)?.Count;

-                var greatBallsCount = allitems.FirstOrDefault(i => i.ItemId == ItemId.ItemGreatBall)?.Count;

-                var ultraBallsCount = allitems.FirstOrDefault(i => i.ItemId == ItemId.ItemUltraBall)?.Count;

-                var masterBallsCount = allitems.FirstOrDefault(i => i.ItemId == ItemId.ItemMasterBall)?.Count;

-                masterBallsCount =

-                    masterBallsCount ?? 0; //return null ATM. need this code to logic check work

-

-                if (pokeBallsCount + greatBallsCount + ultraBallsCount + masterBallsCount <

+                if (PokeBalls <

Can be simplified to one line, if removing extra indents

See above remark.


In PoGo.NecroBot.Logic/Tasks/EvolvePokemonTask.cs<https://github.com/Necrobot-Private/NecroBot/pull/1757#discussion_r119267080>:

> @@ -123,7 +151,9 @@ public class EvolvePokemonTask

                     try

                     {

                         // no cancellationToken.ThrowIfCancellationRequested here, otherwise the lucky egg would be wasted.

-                        var evolveResponse = await session.Client.Inventory.EvolvePokemon(pokemon.Id ,filter== null? ItemId.ItemUnknown: await GetRequireEvolveItem(session ,pokemon.PokemonId, filter.EvolveToPokemonId)).ConfigureAwait(false);

Unneccesary Change( Should be 'TextHere, MoreText' not 'TextHere ,MoreText' )

This is definitely something I would have changed if I’d seen it. I like neat code.


In PoGo.Necrobot.Window/Controls/PlayerInfo.xaml<https://github.com/Necrobot-Private/NecroBot/pull/1757#discussion_r119267730>:

> @@ -110,9 +98,11 @@

                     </MultiBinding>

                 </TextBlock.Text>

             </TextBlock>

-

-

+            <TextBlock Grid.Column="2" Grid.Row="2" x:Name="tb_lukyegg_time" TextWrapping="Wrap" Text="{Binding Lucky_expires, FallbackValue=LuckyEggs: 00m 00s, StringFormat=LuckyEggs: \{0\}, UpdateSourceTrigger=PropertyChanged}" VerticalAlignment="Top" Width="158" Height="16"/>

+            <TextBlock Grid.Column="3" Grid.Row="2" x:Name="tb_incense_time" TextWrapping="Wrap" Text="{Binding Insence_expires, FallbackValue=Insence: 00m 00s, StringFormat=Insence: \{0\}, UpdateSourceTrigger=PropertyChanged}" VerticalAlignment="Top" Width="158" Height="16"/>

+            <Button x:Name="btnCollectPokeCoin" Click="GetPokeCoin_Click" Grid.Column="4" Grid.Row="1" HorizontalAlignment="Center" VerticalAlignment="Center" Margin="23.4,24.4,24,0.2" Grid.RowSpan="2" Width="111">

For this Button, Margin wouldn't really be necessary...

Was not I that added this new code for new info in PlayerInfo except for the EggTimer info(Work in progress)


In PoGo.Necrobot.Window/Controls/PlayerInfo.xaml<https://github.com/Necrobot-Private/NecroBot/pull/1757#discussion_r119267804>:

> @@ -110,9 +98,11 @@

                     </MultiBinding>

                 </TextBlock.Text>

             </TextBlock>

-

-

+            <TextBlock Grid.Column="2" Grid.Row="2" x:Name="tb_lukyegg_time" TextWrapping="Wrap" Text="{Binding Lucky_expires, FallbackValue=LuckyEggs: 00m 00s, StringFormat=LuckyEggs: \{0\}, UpdateSourceTrigger=PropertyChanged}" VerticalAlignment="Top" Width="158" Height="16"/>

+            <TextBlock Grid.Column="3" Grid.Row="2" x:Name="tb_incense_time" TextWrapping="Wrap" Text="{Binding Insence_expires, FallbackValue=Insence: 00m 00s, StringFormat=Insence: \{0\}, UpdateSourceTrigger=PropertyChanged}" VerticalAlignment="Top" Width="158" Height="16"/>

+            <Button x:Name="btnCollectPokeCoin" Click="GetPokeCoin_Click" Grid.Column="4" Grid.Row="1" HorizontalAlignment="Center" VerticalAlignment="Center" Margin="23.4,24.4,24,0.2" Grid.RowSpan="2" Width="111">

I also dont see why 2 Grid Rows would be needed, unless pokecoins go over 9 Digits

Same comment as above.


In PoGo.Necrobot.Window/Model/PlayerInfoModel.cs<https://github.com/Necrobot-Private/NecroBot/pull/1757#discussion_r119267911>:

>

 namespace PoGo.Necrobot.Window.Model

 {

     public class PlayerInfoModel : ViewModelBase

     {

+        private DateTime expires = new DateTime(0);

+        private System.Timers.Timer tmr = new Timer()

In Line 13 System.Timers is already defined, so i don't see it necessary to be referenced twice

I haven’t had a chance to view the new incense/LuckyEgg timers yet.


In RocketBot2/Config/Translations/translation.fr.json<https://github.com/Necrobot-Private/NecroBot/pull/1757#discussion_r119267967>:

>      },

     {

       "Key": "eventPokemonCaptureFailed",

-      "Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | {8}m dist | avec une {9} ({10} restante(s)) | lat: {11} long: {12} | Attaque1: {13} Attaque2: {14}  | Rareté :{15}"

+      "Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | {8}m dist | avec une {9} ({10}  restante(s)). | XP: {11} | lat: {12} long: {13} | Attaque1: {14} Attaque2: {15} | Rareté :{16}"

Same Review Statements recommended as Necrobot Translation File for 'fr'

As mention above I haven’t touched any other translations other than English.


In RocketBot2/Forms/MainForm.cs<https://github.com/Necrobot-Private/NecroBot/pull/1757#discussion_r119268050>:

> @@ -95,6 +95,9 @@ public MainForm(string[] _args)



         private void MainForm_Load(object sender, EventArgs e)

         {

+            this.splitContainer1.SplitterDistance = this.splitContainer1.Width / 100 * 45; // Splits left & right splitter panes @ 45%/55% of the window width

Extra Spaces on Width / 100 and Height / 100 are not necessary

True but it does look neater to me. Clearer definition between var’s


In RocketBot2/Models/GMapMarkerPokestops.cs<https://github.com/Necrobot-Private/NecroBot/pull/1757#discussion_r119268137>:

> @@ -30,7 +30,7 @@ public GMapMarkerPokestops(PointLatLng p, Image image)

         {

             MarkerImage = image;

             Size = MarkerImage.Size;

-            Offset = new Point(-Size.Width/2, -Size.Height);

+            Offset = new Point(-Size.Width / 2, -Size.Height / 2);

Excess Space with Width / 2 and Height / 2

Same Comment as above.


In RocketBot2/Models/GMapMarkerTrainer.cs<https://github.com/Necrobot-Private/NecroBot/pull/1757#discussion_r119268169>:

> @@ -30,7 +30,7 @@ public GMapMarkerTrainer(PointLatLng p, Image image)

         {

             MarkerImage = image;

             Size = MarkerImage.Size;

-            Offset = new Point(-Size.Width/2, -Size.Height/2);

+            Offset = new Point(-Size.Width / 2, -Size.Height / 2);

Excess Space with Width / 2 and Height / 2

Same comment as above.

File: PoGo.NecroBot.CLI/Config/Translations/translation.fr.json (hunk 1 above)
Comment (Furtif) [inline PoGo.NecroBot.CLI/Config/Translations/translation.fr.json]: affter merges default is   {
      "Key": "eventPokemonCaptureSuccess",
      "Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | {8}m dist | with a {9} ({10} left). | XP: {11} | SD: {12} | {13} | lat: {14} long: {15} | Move1: {16} Move2: {17} | Rarity: {18} | Capture Reason: {19} | Shiny: {20} | Form: {21} | Costume: {22} | Sex: {23}"
    },
    {
      "Key": "eventPokemonCaptureFailed",
      "Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | {8}m dist | with a {9} ({10} left). | XP: {11} | lat: {12} long: {13} | Move1: {14} Move2: {15} | Rarity :{16}"
    },
    

seem in translations.cs

File: RocketBot2/Config/Translations/translation.fr.json (hunk 8 above)
Comment (Furtif) [inline RocketBot2/Config/Translations/translation.fr.json]: affter merges default is {
"Key": "eventPokemonCaptureSuccess",
"Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | {8}m dist | with a {9} ({10} left). | XP: {11} | SD: {12} | {13} | lat: {14} long: {15} | Move1: {16} Move2: {17} | Rarity: {18} | Capture Reason: {19} | Shiny: {20} | Form: {21} | Costume: {22} | Sex: {23}"
},
{
"Key": "eventPokemonCaptureFailed",
"Value": "({0}) | ({1}) {2} Lvl: {3} CP: ({4}/{5}) IV: {6}% | Chance: {7}% | {8}m dist | with a {9} ({10} left). | XP: {11} | lat: {12} long: {13} | Move1: {14} Move2: {15} | Rarity :{16}"
},

seem in translations.cs

File: PoGo.NecroBot.CLI/Program.cs (hunk 2 above)
Comment (Furtif) [inline PoGo.NecroBot.CLI/Program.cs]: ready in https://github.com/Necrobot-Private/NecroBot/pull/1784

File: PoGo.NecroBot.Logic/Tasks/CatchNearbyPokemonsTask.cs (hunk 3 above)
Comment (Furtif) [inline PoGo.NecroBot.Logic/Tasks/CatchNearbyPokemonsTask.cs]: ?! https://github.com/Necrobot-Private/NecroBot/pull/1784

File: PoGo.NecroBot.Logic/Tasks/CatchNearbyPokemonsTask.cs (hunk 4 above)
Comment (Furtif) [inline PoGo.NecroBot.Logic/Tasks/CatchNearbyPokemonsTask.cs]: https://github.com/Necrobot-Private/NecroBot/pull/1784 seem

File: PoGo.Necrobot.Window/Model/PlayerInfoModel.cs (hunk 7 above)
Comment (Furtif) [inline PoGo.Necrobot.Window/Model/PlayerInfoModel.cs]: I'm not sure that the timer is ok to change the text every 1seconds

**Comment by Furtif** [comment]
fixes in https://github.com/Necrobot-Private/NecroBot/issues/1784

**Comment by CDAGaming** [comment]
Keep in mind @TheWizard1328 , the review is directed for everyone contributing to the PR 👍

**Comment by Furtif** [comment]
@CDAGaming i m restaured CatchNearbyPokemonsTask.cs

**Comment by Furtif** [comment]
https://github.com/Necrobot-Private/NecroBot/pull/1790 one fix more

**Comment by Furtif** [comment]
... @CDAGaming  Check it out
 Almost ready, some finalizations are missing ...

**Comment by Furtif** [comment]
sans titre

**Comment by CDAGaming** [comment]
Will do a 2nd Review and Personal Test Tomorrow Morning

**Comment by Lord-Haji** [comment]
bug fixes later

**Pull request merged**